swarm/storage/schema.go

```diff
@@ -0,0 +1,6 @@
+package storage
+
+// "hive" is the first formal schema of LevelDB we release together with Swarm 0.3.4
```

Comment:
I think it is unnecessary to use names, the migration functions should be named properly.
"hive" I find especially misleading here.
Migrations is a list of migrate functions to update between versions. Each of which should be applied to bring the found version uptodate.


```golang
// migration function type
type migrateFunc func(*LocalStore) error

// Migrations is the list of migrate functions called to update db schemas
var Migrations = []migrateFunc{
    removeInvalidChunks,
}

func removeInvalidChunks(ls *LocalStore) error {
   return ls.DbStore.Cleanup()
}
```

`GetVersion` and `PutVersion` functions could just work with integers

and current version is always `len(Migrations)`

Reply:
I agree `hive` is not the best name for the first schema, as we use this name elsewhere. Let's rename it.

As per my comment above, I think this style is adding unnecessary complexity for migrations that we don't yet have, and should really be avoiding.

The `names` vs `version` is up for discussion though.

As an example if we mess up a specific version badly, figure it out 1-2 days later when only a bunch of people have upgraded, we can just fix our code, re-deploy, and clean the specific version's LevelDB. This won't be possible with this code, where we have ordering semantics.